contracts/bonding/BondingManager.sol

```diff
@@ -421,20 +417,23 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
         Delegator storage del = delegators[_transcoder];
 
         if (penalty > del.bondedAmount) {
-            // Decrease transcoder's total stake by transcoder's stake
-            transcoderPools.decreaseTranscoderStake(_transcoder, del.bondedAmount);
+            // Decrease delegate's delegated amount
+            delegators[del.delegateAddress].delegatedAmount = delegators[del.delegateAddress].delegatedAmount.sub(del.bondedAmount);
             // Set transcoder's bond to 0 since
             // the penalty is greater than its stake
             del.bondedAmount = 0;
         } else {
-            // Decrease transcoder's total stake by the penalty
-            transcoderPools.decreaseTranscoderStake(_transcoder, penalty);
+            // Decrease delegate's delegated amount
+            delegators[del.delegateAddress].delegatedAmount = delegators[del.delegateAddress].delegatedAmount.sub(penalty);
             // Decrease transcoder's stake
             del.bondedAmount = del.bondedAmount.sub(penalty);
         }
 
-        // Set withdraw round for delegators
-        transcoders[msg.sender].delegatorWithdrawRound = roundsManager().currentRound().add(unbondingPeriod);
+        // Decrease total active stake
+        totalActiveTranscoderStake = totalActiveTranscoderStake.sub(activeTranscoderTotalStake(_transcoder));
```

Comment:
Is this calculation affected by the lines above which decreased the transcoder's "delegated amount"? I'm not sure if the total stake needs to account for the penalty, or if it's just copied from some value which was written down before the penalty was subtracted.

Reply:
It's not because this line is subtracting the transcoder's active stake which was locked in for the round. After the transcoder is slashed none of its stake counts toward the total active stake so we shouldn't need to account for the penalty here - we just need to account for the penalty for the transcoder's delegated amount since delegators will still be bonded to it after it is slashed